Tests/test_MarkovModel.py

```diff
@@ -234,6 +242,254 @@ def test_topcoder5(self):
+        alphabet = "AGTC"
+        p_initial = array([1.0, 0.0])
+        p_transition = array([[0.75, 0.25], [0.25, 0.75]])
+        p_emission = array(
+            [[0.45, 0.36, 0.06, 0.13], [0.24, 0.18, 0.12, 0.46]])
+        markov_model_save = MarkovModel.MarkovModel(
+            states,
+            alphabet,
+            p_initial,
+            p_transition,
+            p_emission)
+
+        handle = StringIO()
+        MarkovModel.save(markov_model_save, handle)
+        handle.seek(0)
+        markov_model_load = MarkovModel.load(handle)
+
+        self.assertEqual(''.join(markov_model_load.states), states)
+        self.assertEqual(''.join(markov_model_load.alphabet), alphabet)
+        self.assertTrue(array_equal(markov_model_load.p_initial, p_initial))
+        self.assertTrue(array_equal
+                        (markov_model_load.p_transition, p_transition))
+        self.assertTrue(array_equal(markov_model_load.p_emission, p_emission))
+
+    def test_train_bw(self):
+        random.seed(0)
+        states = ["0", "1", "2", "3"]
+        alphabet = ["A", "C", "G", "T"]
+        training_data = ["AACCCGGGTTTTTTT", "ACCGTTTTTTT",
+                         "ACGGGTTTTTT", "ACCGTTTTTTTT"]
+
+        output_p_initial = array([0.2275677, 0.29655611,
+                                  0.24993822, 0.22593797])
+        output_p_transition = array(
+            [[5.16919807e-001, 3.65825814e-033, 4.83080193e-001, 9.23220689e-042],
+             [3.65130247e-001,
+              1.00000000e-300,
+              6.34869753e-001,
+              1.00000000e-300],
+             [8.68776164e-001,
+              1.02254304e-034,
+              1.31223836e-001,
+              6.21835051e-047],
+             [3.33333333e-301, 3.33333333e-001, 3.33333333e-301, 6.66666667e-001]])
+        output_p_emission = array(
+            [[2.02593570e-301, 2.02593570e-301, 2.02593570e-301, 1.00000000e+000],
+             [1.00000000e-300,
+              1.00000000e-300,
+              1.00000000e+000,
+              1.09629016e-259],
+             [3.26369779e-301,
+              3.26369779e-301,
+              3.26369779e-301,
+              1.00000000e+000],
+             [3.33333333e-001, 6.66666667e-001, 3.33333333e-301, 3.33333333e-301]])
+
+        markov_model = MarkovModel.train_bw(states, alphabet, training_data)
+        self.assertEqual(''.join(markov_model.states), ''.join(states))
+        self.assertEqual(''.join(markov_model.alphabet), ''.join(alphabet))
+        self.assertTrue(array_equal(
+            around(markov_model.p_initial, decimals=3),
+            around(output_p_initial, decimals=3)))
+        self.assertTrue(array_equal(around(
+            markov_model.p_transition, decimals=3),
+            around(output_p_transition, decimals=3)))
+        self.assertTrue(
+            array_equal(around(
+                        markov_model.p_emission, decimals=3),
+                        around(output_p_emission, decimals=3)))
```

Comment:
The indentation of the first example is the most readable, with the two things being compared on their on lines (self contained).